src/pages/Articles.jsx

```diff
@@ -1,28 +1,49 @@
 import { useEffect, useState } from "react";
+import {useSearchParams} from "react-router-dom"
 import ArticleSummary from "../components/ArticleSummary";
 import Sort from "../components/Sort";
 import { getArticles } from "../utils";
 import "../Articles.css"
 
-const Articles = () => {
+const Articles = ({setQueryString, queryString}) => {
   const [articles, setArticles] = useState([{}]);
   const [isLoading, setIsLoading] = useState(true);
+  const [sort, setSort] = useState("")
+  const [ascDesc, setAscDesc] = useState("")
+  let [searchParams, setSearchParams] = useSearchParams()
+  
+
+  const queryStringBuilder = () => {
+    if(sort && ascDesc){
+      setQueryString(`?sort_by=${sort}&order=${ascDesc}`)
+    }
+    else if(!sort && ascDesc){
+      setQueryString(`?order=${ascDesc}`)
+    }
+    else if (sort && !ascDesc){
+      setQueryString(`?sort_by=${sort}`)
+    }
+    else setQueryString("")
+  }
+
 
   useEffect(() => {
```

Comment:
We may want to set loading to true inside the useEffect if doing another api request.